subprojects/core/src/integTest/groovy/org/gradle/api/internal/changedetection/state/TaskTypeUpToDateIntegrationTest.groovy

```diff
@@ -176,6 +180,33 @@ class TaskTypeUpToDateIntegrationTest extends AbstractIntegrationSpec {
         then: skipped(":copy")
     }
 
+    @Issue("https://github.com/gradle/gradle/issues/9723")
+    def "declaring a task action receiving #incrementalChangesType without declaring outputs is deprecated"() {
+        def input = file('input.txt').createFile()
+        buildFile << """
+            class Task extends DefaultTask {
+                @InputFile File input
+             
+                @TaskAction execute(${incrementalChangesType} inputChanges) {
+                }
+            }   
+            
+            task noOutput(type: Task) {
+                input = file('${input.name}')
+            }
+        """
+
+        when:
+        executer.expectDeprecationWarning()
+        run 'noOutput'
+        then:
+        outputContains("Using the incremental task API without declaring any outputs has been deprecated. This is scheduled to be removed in Gradle 6.0. Please declare output files for your task or use `task.upToDateWhen { true }`.")
```

Comment:
do we usually include the whole message in the tests?

Reply:
We often do, so it is
- easier for the reviewer to see what the actual message is
- the test will fail when we upgrade the base version to `7.0`, since then the message will be different. That way we know what things we need to remove.